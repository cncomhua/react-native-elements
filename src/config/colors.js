import { Platform, StyleSheet } from 'react-native';

export default {
  primary: Platform.select({
    default: '#2196f3', // rgb(33, 150, 243) Android
    ios: '#007aff' // rgb(0, 122, 255)
  }),
  secondary: '#8F0CE8',
  grey0: '#393e42',
  grey1: '#43484d',
  grey2: '#5e6977',
  grey3: '#86939e',
  grey4: '#bdc6cf',
  grey5: '#e1e8ee',
  greyOutline: '#bbb',
  searchBg: '#303337',
  success: Platform.select({
    default: '#4caf50', // Android
    ios: '#4cd964'
  }),
  error: Platform.select({
    default: '#f44336', // Android
    ios: '#ff3b30'
  }),
  warning: Platform.select({
    default: '#ffeb3b', // Android
    ios: '#ffcc00'
  }),
  disabled: 'hsl(208, 8%, 90%)',
  // Darker color if hairlineWidth is not thin enough
  divider: StyleSheet.hairlineWidth < 1 ? '#bcbbc1' : 'rgba(0, 0, 0, 0.12)',
};
