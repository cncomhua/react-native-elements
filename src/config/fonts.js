export default {
  android: {
    regular: {
      fontFamily: 'sans-serif',
      fontWeight: 'regular',
    },
    light: {
      fontFamily: 'sans-serif-light',
      fontWeight: 'regular',
    },
    condensed: {
      fontFamily: 'sans-serif-condensed',
      fontWeight: 'regular',
    },
    condensed_light: {
      fontFamily: 'sans-serif-condensed',
      fontWeight: 'light',
    },
    black: {
      // note(brentvatne): sans-serif-black is only supported on Android 5+,
      // we can detect that here and use it in that case at some point.
      fontFamily: 'sans-serif',
      fontWeight: 'bold',
    },
    thin: {
      fontFamily: 'sans-serif-thin',
      fontWeight: 'regular',
    },
    medium: {
      fontFamily: 'sans-serif-medium',
      fontWeight: 'regular',
    },
    bold: {
      fontFamily: 'sans-serif',
      fontWeight: 'bold',
    },
  },
  default: {},
};
