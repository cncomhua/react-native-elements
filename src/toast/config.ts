import { StylePropsWithMessageType, ToastPosition } from './ToastProvider';

import type { StyleProp, TextProps, TextStyle, ViewStyle } from 'react-native';

export type DefaultConfigType = {
  duration: number;
  maxMessages: number;
  position: ToastPosition;
  containerToastStyle: StyleProp<ViewStyle>;
  containerMessageStyle: StylePropsWithMessageType<ViewStyle>;
  textMessageStyle: StylePropsWithMessageType<TextStyle>;
  textMessageProps: TextProps;
};

const defaultConfig: DefaultConfigType = {
  duration: 2000,
  maxMessages: 5,
  position: ToastPosition.top,
  containerToastStyle: {
    position: 'absolute',
    left: 0,
    right: 0,
    zIndex: 99999,
  },
  containerMessageStyle: {
    margin: 10,
    marginBottom: 5,
    padding: 10,
    borderRadius: 4,
    backgroundColor: '#586ae3',
    shadowColor: 'black',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.15,
    shadowRadius: 5,
    elevation: 6,
    // typed styles
    info: {},
    error: {},
    warning: {},
    success: {},
  },
  textMessageStyle: {
    // typed styles
    info: {},
    error: {},
    warning: {},
    success: {},
  },
  textMessageProps: {},
};

export default defaultConfig;
