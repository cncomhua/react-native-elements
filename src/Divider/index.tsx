import { Divider, DividerProps } from './Divider';
import { withTheme } from '../config';

export { Divider };
export type { DividerProps };
/**
 * RNE Divider
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/divider)
 */
export default withTheme(Divider, 'Divider');
