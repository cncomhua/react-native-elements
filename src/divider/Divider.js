import React from 'react';
import { View, ViewPropTypes, StyleSheet } from 'react-native';
import colors from '../config/colors';

let styles = {};

const Divider = ({ style }) => (
  <View style={[styles.container, style && style]} />
);

Divider.propTypes = {
  style: ViewPropTypes.style,
};

styles = StyleSheet.create({
  container: {
    height: 1,
    backgroundColor: colors.grey5,
  },
});

export default Divider;
