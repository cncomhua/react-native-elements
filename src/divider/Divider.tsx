import React from 'react';
import { View, StyleSheet, ViewProps } from 'react-native';
import { withTheme } from '../config';
import theme from '../config/theme';
import { RneFunctionComponent } from '../helpers';

export type DividerProps = ViewProps & {
  style?: object | any[];
  inset?: boolean;
  insetType?: 'left' | 'right' | 'middle';
  orientation?: 'horizontal' | 'vertical';
};

const Divider: RneFunctionComponent<DividerProps> = ({
  inset = false,
  insetType = 'left',
  orientation = 'horizontal',
  style,
  theme,
  ...rest
}) => (
  <View
    style={StyleSheet.flatten([
      styles.divider,
      style,
      inset &&
        (insetType === 'left'
          ? styles.leftInset
          : insetType === 'right'
          ? styles.rightInset
          : { ...styles.leftInset, ...styles.rightInset }),
      orientation === 'vertical' && styles.vertical,
    ])}
    {...rest}
  />
);

const styles = StyleSheet.create({
  divider: {
    backgroundColor: theme?.colors?.divider,
    height: StyleSheet.hairlineWidth,
  },
  leftInset: {
    marginLeft: 72,
  },
  rightInset: {
    marginRight: 72,
  },
  vertical: {
    borderRightWidth: StyleSheet.hairlineWidth,
    borderRightColor: theme?.colors?.divider,
    height: 'auto',
    alignSelf: 'stretch',
  },
});

export { Divider };
export default withTheme(Divider, 'Divider');
