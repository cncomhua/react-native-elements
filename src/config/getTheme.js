import { StyleSheet, Platform } from 'react-native';
import base from './colors';
import fonts from './fonts';
import normalize from '../helpers/normalizeText';

export default function getTheme(theme) {
  const colors = {
    ...base,
    ...theme,
  };

  return {
    colors,
    badge: StyleSheet.create({
      container: {
        flexDirection: 'row',
      },
      badge: {
        padding: 12,
        paddingTop: 3,
        paddingBottom: 3,
        backgroundColor: colors.grey1,
        borderRadius: 20,
        alignItems: 'center',
        justifyContent: 'center',
      },
      text: {
        fontSize: 14,
        color: 'white',
      },
    }),
    card: StyleSheet.create({
      container: {
        backgroundColor: 'white',
        borderColor: colors.grey5,
        borderWidth: 1,
        padding: 15,
        margin: 15,
        marginBottom: 0,
        ...Platform.select({
          ios: {
            shadowColor: 'rgba(0,0,0, .2)',
            shadowOffset: { height: 0, width: 0 },
            shadowOpacity: 1,
            shadowRadius: 1,
          },
          android: {
            elevation: 1,
          },
        }),
      },
      featuredTitle: {
        fontSize: normalize(18),
        marginBottom: 8,
        color: 'white',
        ...Platform.select({
          ios: {
            fontWeight: '800',
          },
          android: {
            ...fonts.android.black,
          },
        }),
      },
      featuredSubtitle: {
        fontSize: normalize(13),
        marginBottom: 8,
        color: 'white',
        ...Platform.select({
          ios: {
            fontWeight: '400',
          },
          android: {
            ...fonts.android.black,
          },
        }),
      },
      wrapper: {
        backgroundColor: 'transparent',
      },
      divider: {
        marginBottom: 15,
      },
      cardTitle: {
        fontSize: normalize(14),
        ...Platform.select({
          ios: {
            fontWeight: 'bold',
          },
          android: {
            ...fonts.android.black,
          },
        }),
        textAlign: 'center',
        marginBottom: 15,
        color: colors.grey1,
      },
      imageCardTitle: {
        marginTop: 15,
      },
      overlayContainer: {
        flex: 1,
        alignItems: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.2)',
        alignSelf: 'stretch',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
      },
    }),
    checkbox: StyleSheet.create({
      wrapper: {
        flexDirection: 'row',
        alignItems: 'center',
      },
      container: {
        margin: 5,
        marginLeft: 10,
        marginRight: 10,
        backgroundColor: '#fafafa',
        borderColor: '#ededed',
        borderWidth: 1,
        padding: 10,
        borderRadius: 3,
      },
      text: {
        marginLeft: 10,
        marginRight: 10,
        color: colors.grey1,
        ...Platform.select({
          ios: {
            fontWeight: 'bold',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
    }),
    divider: StyleSheet.create({
      container: {
        height: 1,
        backgroundColor: colors.grey5,
      },
    }),
    header: StyleSheet.create({
      innerContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'flex-end',
      },
      outerContainer: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        backgroundColor: '#fff',
        borderBottomColor: '#f2f2f2',
        borderBottomWidth: 1,
        padding: 15,
        height: 70,
      },
    }),
    icon: StyleSheet.create({
      button: {
        margin: 7,
      },
      raised: {
        ...Platform.select({
          ios: {
            shadowColor: 'rgba(0,0,0, .4)',
            shadowOffset: { height: 1, width: 1 },
            shadowOpacity: 1,
            shadowRadius: 1,
          },
          android: {
            elevation: 2,
          },
        }),
      },
    }),
    input: StyleSheet.create({
      container: {
        flexDirection: 'row',
        borderBottomWidth: 1,
        borderColor: 'rgba(171, 189, 219, 1)',
        alignItems: 'center',
      },
      iconContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        marginLeft: 15,
      },
      input: {
        alignSelf: 'center',
        color: 'black',
        fontSize: 18,
        marginLeft: 10,
      },
      error: {
        color: '#FF2D00',
        margin: 5,
        fontSize: 12,
      },
    }),
    list: StyleSheet.create({
      listContainer: {
        marginTop: 20,
        borderTopWidth: 1,
        borderColor: colors.greyOutline,
        backgroundColor: colors.white,
      },
    }),
    listItem: StyleSheet.create({
      avatar: {
        width: 34,
        height: 34,
      },
      container: {
        paddingTop: 10,
        paddingRight: 10,
        paddingBottom: 10,
        borderBottomColor: colors.greyOutline,
        borderBottomWidth: 1,
        backgroundColor: 'transparent',
      },
      wrapper: {
        flexDirection: 'row',
        marginLeft: 10,
      },
      iconStyle: {
        justifyContent: 'center',
        alignItems: 'center',
      },
      icon: {
        marginRight: 8,
      },
      title: {
        fontSize: normalize(14),
        color: colors.grey1,
      },
      subtitle: {
        color: colors.grey3,
        fontSize: normalize(12),
        marginTop: 1,
        ...Platform.select({
          ios: {
            fontWeight: '600',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
      titleSubtitleContainer: {
        justifyContent: 'center',
        flex: 1,
      },
      chevronContainer: {
        flex: 0.15,
        alignItems: 'flex-end',
        justifyContent: 'center',
      },
      switchContainer: {
        flex: 0.15,
        alignItems: 'flex-end',
        justifyContent: 'center',
        marginRight: 5,
      },
      rightTitleContainer: {
        flex: 1,
        alignItems: 'flex-end',
        justifyContent: 'center',
      },
      rightTitleStyle: {
        marginRight: 5,
        color: colors.grey4,
      },
      textInputStyle: {
        height: 20,
        textAlign: 'right',
      },
    }),
    pricingCard: StyleSheet.create({
      container: {
        margin: 15,
        marginBottom: 15,
        backgroundColor: 'white',
        borderColor: colors.grey5,
        borderWidth: 1,
        padding: 15,
        ...Platform.select({
          ios: {
            shadowColor: 'rgba(0,0,0, .2)',
            shadowOffset: { height: 1, width: 0 },
            shadowOpacity: 0.5,
            shadowRadius: 0.5,
          },
          android: {
            elevation: 1,
          },
        }),
      },
      wrapper: {
        backgroundColor: 'transparent',
      },
      pricingTitle: {
        textAlign: 'center',
        color: colors.primary,
        fontSize: normalize(30),
        ...Platform.select({
          ios: {
            fontWeight: '800',
          },
          android: {
            ...fonts.android.black,
          },
        }),
      },
      pricingPrice: {
        textAlign: 'center',
        marginTop: 10,
        marginBottom: 10,
        fontSize: normalize(40),
        ...Platform.select({
          ios: {
            fontWeight: '700',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
      pricingInfo: {
        textAlign: 'center',
        marginTop: 5,
        marginBottom: 5,
        color: colors.grey3,
        ...Platform.select({
          ios: {
            fontWeight: '600',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
      button: {
        marginTop: 15,
        marginBottom: 10,
      },
    }),
    search: StyleSheet.create({
      container: {
        borderTopWidth: 1,
        borderBottomWidth: 1,
        borderBottomColor: '#000',
        borderTopColor: '#000',
        backgroundColor: colors.grey0,
      },
      containerLight: {
        backgroundColor: colors.grey5,
        borderTopColor: '#e1e1e1',
        borderBottomColor: '#e1e1e1',
      },
      icon: {
        backgroundColor: 'transparent',
        position: 'absolute',
        top: 15.5,
        ...Platform.select({
          android: {
            top: 20,
          },
        }),
      },
      loadingIcon: {
        backgroundColor: 'transparent',
        position: 'absolute',
        right: 16,
        top: 13,
        ...Platform.select({
          android: {
            top: 18,
          },
        }),
      },
      input: {
        paddingLeft: 26,
        paddingRight: 19,
        margin: 8,
        borderRadius: 3,
        overflow: 'hidden',
        backgroundColor: colors.searchBg,
        fontSize: normalize(14),
        color: colors.grey3,
        height: 40,
        ...Platform.select({
          ios: {
            height: 30,
          },
          android: {
            borderWidth: 0,
          },
        }),
      },
      inputLight: {
        backgroundColor: colors.grey4,
      },
      searchIcon: {
        left: 16,
      },
      clearIcon: {
        right: 16,
      },
    }),
  };
}
