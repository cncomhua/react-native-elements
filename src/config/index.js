import BackgroundImage from './BackgroundImage';
import colors from './colors';
import ThemeProvider from './ThemeProvider';
import getTheme from './getTheme';
import ViewPropTypes from './ViewPropTypes';
import fonts from './fonts';

export {
  BackgroundImage,
  colors,
  fonts,
  ThemeProvider,
  getTheme,
  ViewPropTypes,
};
