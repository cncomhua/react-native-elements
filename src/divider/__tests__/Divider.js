import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Divider from '../Divider';
import { getTheme } from '../../config';

const options = {
  context: { theme: getTheme() },
  childContextTypes: { theme: PropTypes.object },
};

describe('Divider Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Divider />, options);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with style', () => {
    const component = shallow(
      <Divider style={{ backgroundColor: 'blue' }} />,
      options
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    expect(component.props().style.length).toBe(2);
    expect(component.props().style[1].backgroundColor).toBe('blue');
  });
});
