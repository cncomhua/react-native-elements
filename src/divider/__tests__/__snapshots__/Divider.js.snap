// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should apply values from theme 1`] = `
<View
  replaceTheme={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "height": 0.5,
    }
  }
  updateTheme={[Function]}
/>
`;

exports[`Divider Component should pass view properties 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#bcbbc1",
      "height": 0.5,
    }
  }
  testID="testing"
/>
`;

exports[`Divider Component should render with inset true and insetType left 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#bcbbc1",
      "height": 0.5,
      "marginLeft": 72,
    }
  }
/>
`;

exports[`Divider Component should render with inset true and insetType middle 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#bcbbc1",
      "height": 0.5,
      "marginLeft": 72,
      "marginRight": 72,
    }
  }
/>
`;

exports[`Divider Component should render with inset true and insetType right 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#bcbbc1",
      "height": 0.5,
      "marginRight": 72,
    }
  }
/>
`;

exports[`Divider Component should render with orientation vertical 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#bcbbc1",
      "borderRightColor": "#bcbbc1",
      "borderRightWidth": 0.5,
      "height": "auto",
    }
  }
/>
`;

exports[`Divider Component should render with style 1`] = `
<View
  style={
    Object {
      "backgroundColor": "blue",
      "height": 0.5,
    }
  }
/>
`;

exports[`Divider Component should render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#bcbbc1",
      "height": 0.5,
    }
  }
/>
`;
